src/pmi/src/pmi_v2.c

```diff
@@ -365,9 +371,11 @@ PMI_API_PUBLIC int PMI2_KVS_Fence(void)
     struct PMIU_cmd pmicmd;
     PMIU_cmd_init(&pmicmd, USE_WIRE_VER, "kvs-fence");
 
-    pmi_errno = PMIU_cmd_get_response(PMI_fd, &pmicmd, "kvs-fence-response");
-    PMIU_ERR_POP(pmi_errno);
-    PMI2_CHK_RC_ERRMSG(&pmicmd);
+    if (PMI_initialized > SINGLETON_INIT_BUT_NO_PM) {
+        pmi_errno = PMIU_cmd_get_response(PMI_fd, &pmicmd, "kvs-fence-response");
+        PMIU_ERR_POP(pmi_errno);
+        PMI2_CHK_RC_ERRMSG(&pmicmd);
+    }
```

Comment:
I believe this is no longer necessary now that we skips barrier in single process?

Reply:
Yes, we can remove this change.